tests/testthat/test-csv.R

```diff
@@ -340,6 +347,19 @@ test_that("remaining_columns_to_read() works", {
   expect_equal(remaining_columns_to_read(NULL, NULL, c("a", "b", "c")), c("a", "b", "c"))
   expect_equal(remaining_columns_to_read(c("a", "b", "c"), c("a", "b", "c"), NULL), "")
   expect_equal(remaining_columns_to_read(c("a", "b", "c"), NULL, c("a", "b", "c")), c("a", "b", "c"))
+
+  # with vector and matrix variables
+  b_vec <- paste0("b[", 1:4, "]")
+  c_mat <- paste0("c[", c(1,2,1,2), ",", c(1,1,2,2), "]")
+  all_vars <- c("a", b_vec, c_mat)
+  expect_equal(
+    remaining_columns_to_read(c("a", "b[2]"), c("a", c_mat), all_vars),
+    "b[2]"
+  )
+  expect_equal(
+    remaining_columns_to_read(c("a", "b", "c"), c(paste0("b[", c(1,4), "]"), "c[1,2]"), all_vars),
+    c("a", "b[2]", "b[3]", "c[1,1]", "c[2,1]", "c[2,2]")
+  )
```

Comment:
Good call. I just pushed a commit that makes `remaining_columns_to_read()` call  `unique()` before returning. Now it will be

```r
  expect_equal(
    remaining_columns_to_read(c("a", "b[2]", "b"), c("a", c_mat), all_vars),
    c("b[2]", "b[1]", "b[3]", "b[4]")
  )
```

where the elements of `b` are out of order because `b[2]` was asked for before `b` (i.e., the second `b[2]` is removed). This is a bit weird but I think probably the right behavior if the input is `c("b[2]", "b")`.

Reply:
Yes, its probably weird but as requested. Will merge once tests pass.